### PR TITLE
Rover: windvane rework and add sailboat autotest

### PR DESCRIPTION
**Pull request by IamPete1**
This reworks the windvane calculations. The main change is to store the apparent wind in body frame this also applys the filters to at the end of caculations. Currently we would filter the apparent wind and then use that filtered value with the current GPS speed. This change means that the GPS speed and wind speed are taken at a single instant, the resulting true wind is then filtered. A new filter param is added for the true wind. This allows is to use a 'fast' filter on the apparent wind that is used to trim the sail and a slow filter on the true wind that is used to navigate the boat.

This also moves WNDVN_RC_IN_NO to a RCxOPTION.

Also adds a very basic autotest for sailboats. Can be run with: `./Tools/autotest/autotest.py build.Rover test.Sailboat`

Oh and it moves wind logging to the windvane lib in a new `WIND` log, and logs more info, sail outputs are still in the `SAIL` log.

@srmainwaring @arikrupnik

### CODE REVIEW

**Comment by arikrupnik** [comment]
Thank you for doing this, @IamPete1 . As long as you're adding autotests, is there a plan to have unit tests for Rover? I would love to include tests for my code, something like mock relative wind x, existing servo positions y, confirm mast rotation = z.

**Comment by IamPete1** [comment]
@arikrupnik no plans for unit tests as such, in general the auto tests are a bit more holistic, this one just checks it can get round a mission.

**Comment by peterbarker** [comment]
@arikrupnik hard to wedge unit tests in for the sort of thing you describe - but that's probably all doable in a regression test.

File: libraries/AP_WindVane/AP_WindVane_Backend.h [hunk 1]
```diff
@@ -38,16 +38,7 @@ class AP_WindVane_Backend
...
-    void direction_update_frontend(float apparent_angle_ef);
 
     AP_WindVane &_frontend;
 
-private:
-    // low pass filters of direction and speed
-    LowPassFilterFloat _dir_sin_filt = LowPassFilterFloat(2.0f);
-    LowPassFilterFloat _dir_cos_filt = LowPassFilterFloat(2.0f);
-    LowPassFilterFloat _speed_filt = LowPassFilterFloat(2.0f);
-    LowPassFilterFloat _tack_filt = LowPassFilterFloat(TACK_FILT_CUTOFF);
```
Comment (IamPete1) [inline libraries/AP_WindVane/AP_WindVane_Backend.h:52]: use TACK_FILT_CUTOFF

File: Rover/Log.cpp [hunk 2]
```diff
@@ -134,36 +134,27 @@ void Rover::Log_Write_Sail()
...
-    logger.Write("SAIL", "TimeUS,WndDrTru,WndDrApp,WndSpdTru,WndSpdApp,MainOut,WingOut,VMG",
-                        "shhnn%%n", "F0000000", "Qfffffff",
+    logger.Write("SAIL", "TimeUS,Tack,TackThr,MainOut,WingOut,VMG",
+                        "s-d%%n", "F00000", "QBffff",
                         AP_HAL::micros64(),
-                        (double)wind_dir_abs,
-                        (double)wind_dir_rel,
-                        (double)wind_speed_true,
-                        (double)wind_speed_apparent,
+                        (double)current_tack,
```
Comment (IamPete1) [inline Rover/Log.cpp:156]: this should not be a double

Comment (arikrupnik) [reply Rover/Log.cpp:156]: None of this should be here at all. I didn't touch logging. I thought I had cleaned up the merge mess I'd made, but apparently not.

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**